crates/weaver_live_check/README.md

```diff
@@ -0,0 +1,239 @@
+
+Beyond the fundamentals, external `Advisors` can be defined in Rego policies. The OpenTelemetry Semantic Conventions rules are included out-of-the-box by default. They provide `Advice` on name-spacing and formatting aligned with the standard. These default policies can be overridden at the command line with your own.
+
+### Advice
+
+As mentioned, a list of `Advice` is returned in the report for each sample entity. The snippet below shows `Advice` from two `Advisors`. A builtin is providing `missing_attribute` and a default Otel Rego policy is providing `extends_namespace`. The fields of `Advice` are intended to be used like so:
+
+- `advisory`: _string_ - one of `violation`, `improvement` or `information` with that order of precedence. Weaver will return with a non-zero exit-code if there is any `violation` in the report.
+- `key`: _string_ - a simple machine readable string key to represent the advice type
+- `message`: _string_ - a verbose string describing the advice
+- `value`: _any_ - a pertinent entity associated with the advice
+
+```json
+{
+  "all_advice": [
+    {
+      "advisory": "violation",
+      "key": "missing_attribute",
+      "message": "Does not exist in the registry",
+      "value": "aws.s3.extension.name"
+    },
+    {
+      "advisory": "information",
+      "key": "extends_namespace",
+      "message": "Extends existing namespace",
+      "value": "aws.s3"
+    }
+  ],
+  "highest_advisory": "violation",
+  "sample_attribute": {
+    "name": "aws.s3.extension.name",
+    "type": "string",
+    "value": "foo"
+  }
+}
+```
+
+### Custom advisors
+
+Use the `--advice-policies` command line option to provide a path to a directory containing Rego policies with the `advice` package name. Here's a very simple example that rejects any attribute name containing the string "test":
+
+```rego
+package advice
+
+import rego.v1
+
+# checks attribute name contains the word "test"
+deny contains make_advice(key, advisory, value, message) if {
+	contains(input.name, "test")
+	key := "contains_test"
+	advisory := "violation"
+	value := input.name
+	message := "Name must not contain 'test'"
+}
+
+make_advice(key, advisory, value, message) := {
+	"type": "advice",
+	"key": key,
+	"advisory": advisory,
+	"value": value,
+	"message": message,
+}
+```
+
+`input` contains the sample entity for assessment. `data` contains a structure derived from the supplied `Registry`. A jq preprocessor takes the `Registry` (and maps for attributes and templates) to produce the `data` for the policy. If the jq is simply `.` this will passthrough as-is. Preprocessing is used to improve Rego performance and to simplify policy definitions. With this model `data` is processed once whereas the Rego policy runs for every sample entity as it arrives in the stream. Preprocessing for the default otel policies lead to a ~10x speed-up.
```

Comment:
This is a nit beyond the scope of your PR, but we probably need a link here to the actual *schema* of the JSON for `input` and `data`.